Source/Utils/Inc/Utils/Event/WASMEventDispatcher.h

```diff
@@ -24,7 +24,45 @@ enum class EWASMEventDispatcherTarget : int8_t
 
 // -----------------------------------------------------------------------------
 
-std::shared_ptr< EventDispatcher > MakeWASMEventDispatcher( const EWASMEventDispatcherTarget Target );
+struct WASMEventHandler
+{
+	template< class TEvent >
+	static void FireEvent( EventDispatcher< WASMEventHandler >& Dispatcher, const TEvent& Event )
+	{
+		Dispatcher.FireEvent( GetEventId< TEvent >(), Event );
```

Comment:
Do we want a static assert here to ensure TEvent is a `SerialisableEvent`?  The code you call into with `Dispatcher.FireEvent` has a dynamic cast to `ISerialisable` with a log message on nullptr (which feels more correct with regards to where the error actually is), but I'm wondering if we maybe want to enforce that in advance?

Reply:
Yep that's a good idea, will add that